### PR TITLE
Fix/dish creation

### PR DESCRIPTION
**Pull request by AlanDavd**
Fix creation of dish with an unavailable category Id

- Import MenuCategory into MenuDishesService
- Validate that the category id already exists to create a dish

### CODE REVIEW

File: src/modules/menu-category/menu-category.module.ts [hunk 1]
```diff
@@ -8,5 +8,6 @@ import { MenuCategoryRepository } from './menu-category.respository';
   imports: [TypeOrmModule.forFeature([MenuCategoryRepository])],
   controllers: [MenuCategoryController],
   providers: [MenuCategoryService],
+  exports: [MenuCategoryService],
```
Comment (jcmexdev) [inline src/modules/menu-category/menu-category.module.ts]: why you exports this service?, you can add it in the menu-dishes module like the other modules under typeorm.forFeature function for example:
imports: [
    TypeOrmModule.forFeature([
      OrderDetailRepository,
      OrderRepository,
      UserRepository,
      MenuDishesRepository,
    ]),
  ],

File: src/modules/menu-dishes/menu-dishes.module.ts [hunk 2]
```diff
@@ -3,9 +3,13 @@ import { TypeOrmModule } from '@nestjs/typeorm';
 import { MenuDishesRepository } from './menu-dishes.repository';
 import { MenuDishesService } from './menu-dishes.service';
 import { MenuDishesController } from './menu-dishes.controller';
+import { MenuCategoryModule } from '../menu-category/menu-category.module';
 
 @Module({
-  imports: [TypeOrmModule.forFeature([MenuDishesRepository])],
+  imports: [
+    MenuCategoryModule,
+    TypeOrmModule.forFeature([MenuDishesRepository]),
```
Comment (jcmexdev) [inline src/modules/menu-dishes/menu-dishes.module.ts]: you can add the menu category repository here and you don't need the export in menu category module:
TypeOrmModule.forFeature([MenuDishesRepository, MenuCategorRepository ]),

File: src/modules/menu-dishes/menu-dishes.service.ts [hunk 3]
```diff
@@ -1,14 +1,22 @@
...
 import { MenuDishesResponseDto } from './dtos/menu-dishes-response.dto';
 import { MenuDishUpdateDto } from './dtos/menu-dishes-update.dto';
+import { MenuCategoryService } from '../menu-category/menu-category.service';
 
 @Injectable()
 export class MenuDishesService {
-  constructor(private readonly _menuDishesRepository: MenuDishesRepository) {}
+  constructor(
+    private readonly _menuDishesRepository: MenuDishesRepository,
+    private readonly _menuCategoriesService: MenuCategoryService,
```
Comment (jcmexdev) [inline src/modules/menu-dishes/menu-dishes.service.ts]: here we should use the repositories like the other models instead use service directly to have a same conventions

**Review by jcmexdev — CHANGES_REQUESTED** [review]

**Review by jcmexdev — APPROVED** [review]
Good Job!!

**Review by jlazo — APPROVED** [review]

**Pull request merged**